src/lib.rs

```diff
@@ -857,10 +857,33 @@ impl BumpFeeTxBuilder {
     }
 }
 
-fn generate_mnemonic(word_count: WordCount) -> Result<String, BdkError> {
-    let mnemonic: GeneratedKey<_, BareCtx> =
-        Mnemonic::generate((word_count, Language::English)).unwrap();
-    Ok(mnemonic.to_string())
+struct Mnemonic {
+    internal: BdkMnemonic,
+}
+
+impl Mnemonic {
+    fn new(word_count: WordCount) -> Self {
+        let generated_key: GeneratedKey<_, BareCtx> =
+            BdkMnemonic::generate((word_count, Language::English)).unwrap();
+        let mnemonic = BdkMnemonic::parse_in(Language::English, generated_key.to_string()).unwrap();
+        Mnemonic { internal: mnemonic }
+    }
+
+    fn from_string(mnemonic: String) -> Result<Self, BdkError> {
+        BdkMnemonic::parse_in(Language::English, mnemonic)
+            .map(|x| Mnemonic { internal: x })
```

Comment:
this is a small nit.. but can you use `m` here? similar to how you used `e` for error below.
```suggestion
            .map(|m| Mnemonic { internal: m })
```